Lib/distutils/util.py

```diff
@@ -36,8 +36,18 @@ def get_host_platform():
 
     """
     if os.name == 'nt':
-        if 'GCC' in sys.version:
-            return 'mingw'
+        if 'gcc' in sys.version.lower():
+            if 'ucrt' in sys.version.lower():
+                if 'amd64' in sys.version.lower():
+                    return 'mingw_x86_64_ucrt'
+                return 'mingw_i686_ucrt'
+            if 'clang' in sys.version.lower():
+                if 'amd64' in sys.version.lower():
+                    return 'mingw_x86_64_clang'
+                return 'mingw_i686_clang'
```

Comment:
```suggestion
                if '(arm64)' in sys.version.lower():
                    return 'mingw_aarch64_clang'
                return 'mingw_i686_clang'
```

???  arm32 isn't built here, so don't know if you want to handle that for completeness or not.

Reply:
Yeah, I missed about arm here. If you could tell me what I should do for arm32 I'll do that.

Reply:
In general the triple should be armv7-w64-mingw32 if that helps.  I have no real preference, as nobody really seems that excited about 32-bit arm support.

Reply:
In that case, we can handle arm32 when it exists.